apps/OpenSpace/ext/launcher/include/sgctedit/windowcontrol.h

```diff
@@ -0,0 +1,176 @@
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#ifndef __OPENSPACE_UI_LAUNCHER___WINDOWCONTROL___H__
+#define __OPENSPACE_UI_LAUNCHER___WINDOWCONTROL___H__
+
+#include <QCheckBox>
+#include <QColor>
+#include <QComboBox>
+#include <QFrame>
+#include <QIntValidator>
+#include <QLabel>
+#include <QLayout>
+#include <QLineEdit>
+#include <QPainter>
+#include <QPainterPath>
+#include <QPoint>
+#include <QPushButton>
+#include <QStandardItemModel>
+#include <QVector>
+#include <QWidget>
+
+#include <vector>
+#include <iostream>
+#include <sgct/config.h>
+
+
+class WindowControl : public QWidget
+{
+Q_OBJECT
+public:
+    explicit WindowControl(unsigned int nMonitors, unsigned int monitorIndex,
+        unsigned int windowIndex, std::vector<QRect>& monitorDims,
+        const QString* winColors, QWidget *parent = nullptr);
+    ~WindowControl();
+    void setDimensions(const QRectF& dimensions);
+    void setWindowChangeCallback(std::function<void(int, int, const QRectF&)> cb);
+    void setWebGuiChangeCallback(std::function<void(unsigned int)> cb);
+    void showWindowLabel(const bool show);
+    QVBoxLayout* initializeLayout();
+    QRectF& dimensions();
+    QLineEdit* lineEditSizeWidth();
+    QLineEdit* lineEditSizeHeight();
+    QLineEdit* lineEditSizeOffsetX();
+    QLineEdit* lineEditSizeOffsetY();
+    QCheckBox* checkBoxWindowDecor();
+    QCheckBox* checkBoxWebGui();
+    QCheckBox* checkBoxSpoutOutput();
+    std::string windowName();
+    sgct::ivec2 windowSize();
+    sgct::ivec2 windowPos();
+    bool isDecorated();
+    bool isSpoutSelected();
+    bool isGuiWindow();
+    void enableGuiWindowSelection(bool enabled);
+    void uncheckWebGuiOption();
+    int projectionSelectedIndex();
+    int qualitySelectedIndex();
+    int qualitySelectedValue();
+    unsigned int monitorNum();
+    float fovH();
+    float fovV();
+    float heightOffset();
+    enum ProjectionIndeces : unsigned int {
+        Planar = 0,
+        Fisheye,
+        Spherical_Mirror,
+        Cylindrical,
+        Equirectangular
+    };
+    std::string ProjectionTypeNames[5] = {"Planar", "Fisheye", "Spherical Mirror",
```

Comment:
These values are only used in the initialization of the `_projectionTypes`.  Moving the `_projectionTypes` definition into the cpp file means that we don't need to explicitly list the types here, but it can all happen in the cpp instead.  Same for the quality